assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -155,14 +172,106 @@ export const reducer = createReducer( ( state, { type, payload } ) => {
 			break;
 		}
 
+		case RECEIVE_QUEUED_NOTIFICATIONS: {
+			state.queuedNotifications = payload.queuedNotifications;
+			break;
+		}
+
 		default:
 			break;
 	}
 } );
 
-export const resolvers = {};
+export const resolvers = {
+	*getQueuedNotifications( viewContext ) {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const notifications = registry
+			.select( CORE_NOTIFICATIONS )
+			.getNotifications();
+
+		// Wait for all dismissed items to be available before filtering.
+		yield Data.commonActions.await(
+			registry.resolveSelect( CORE_USER ).getDismissedItems()
+		);
+
+		const filteredNotifications = Object.values( notifications ).filter(
+			( notification ) => {
+				if ( ! notification.viewContexts.includes( viewContext ) ) {
+					return false;
+				}
+
+				if (
+					!! notification.isDismissible &&
+					registry
+						.select( CORE_NOTIFICATIONS )
+						.isNotificationDismissed( notification.id )
+				) {
+					return false;
+				}
+
+				return true;
+			}
+		);
+
+		const checkRequirementsResults = yield Data.commonActions.await(
+			Promise.all(
+				filteredNotifications.map( async ( { checkRequirements } ) => {
+					if ( typeof checkRequirements === 'function' ) {
+						try {
+							return await checkRequirements( registry );
+						} catch ( e ) {
+							return false; // Prevent `Promise.all()` from being rejected for a single failed promise.
+						}
+					}
+
+					return true;
+				} )
+			)
+		);
+
+		const queuedNotifications = filteredNotifications.filter(
+			( _, i ) => !! checkRequirementsResults[ i ]
+		);
+
+		queuedNotifications.sort( ( a, b ) => {
+			return a.priority - b.priority;
+		} );
+
+		yield actions.receiveQueuedNotifications( queuedNotifications );
```

Comment:
@jimmymadon Everything up to this point might be a bit cleaner to extract to a control, that way we could trigger a re-evaluation of the queue directly if needed, not only through the resolver here.